aipg/rag/service.py

```
@@ -0,0 +1,84 @@
+
+        logger.info(
+            f"RAG search completed: no matching projects found for topic '{topic}'"
+        )
+        return []
+
+    def save(self, topic: str, micro_project: Project) -> None:
+        logger.info(f"RAG save initiated for topic: '{topic}'")
+
+        embeddings = self.embedder.embedding_processor([topic])
+        if not embeddings:
+            raise RuntimeError(f"Failed to generate embedding for topic: '{topic}'")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)